pages/api/issues.ts

```diff
@@ -19,24 +30,52 @@ export default async function handler(
         if (
           'issue_type_id' in body &&
           typeof body.issue_type_id === 'number' &&
-          'id' in body &&
-          typeof body.id === 'string'
+          'tree_id' in body &&
+          typeof body.tree_id === 'string'
         ) {
-          const { id, issue_type_id } = body
-          // make call to supabase using SERVICE_ROLE_KEY
-          // dont expose SERVICE_ROLE_KEY to client
+          // should make a call to login and get a token
+          // with that token cunstruct a new authenticated client
+          // and make a request with that to post an issue
+
+          const { data: loginData, error: loginError } = await postgrest.rpc(
+            'login',
+            {
+              username: ml_pgrest_username,
+              pass: ml_pgrest_pass,
+            }
+          )
+
+          if (loginError) {
+            return res.status(401).json({ error: 'internal server error' })
```

Comment:
To avoid confusion, we should match the message to the status. So I would suggest to supply the error message "Unauthorized" (also to the response 3 lines below)

Reply:
Jupp. You are right. I'll fix this.